save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -245,87 +261,37 @@ class WarnPlugin(
      * 1) reading the file
      * 2) for each line get the warning
      */
+    @Suppress("AVOID_NULL_CHECKS")
     private fun Path.collectWarningsWithLineNumbers(
-        warnPluginConfig: WarnPluginConfig,
-        generalConfig: GeneralConfig
-    ): List<Warning> {
-        val linesFile = fs.readLines(this)
-        return generalConfig.expectedWarningsEndPattern?.let {
-            collectionMultilineWarnings(
-                warnPluginConfig,
-                generalConfig,
-                linesFile,
-                this,
-            )
-        }
-            ?: run {
-                collectionSingleWarnings(
-                    warnPluginConfig,
-                    generalConfig,
-                    linesFile,
-                    this,
-                )
-            }
-    }
-
-    private fun collectionMultilineWarnings(
         warnPluginConfig: WarnPluginConfig,
         generalConfig: GeneralConfig,
-        linesFile: List<String>,
-        file: Path,
-    ): List<Warning> = linesFile.mapIndexed { index, line ->
-        val newLineAndMessage = line.getLineNumberAndMessage(
-            generalConfig.expectedWarningsPattern!!,
-            generalConfig.expectedWarningsEndPattern!!,
-            generalConfig.expectedWarningsMiddlePattern!!,
-            warnPluginConfig.messageCaptureGroupMiddle!!,
-            warnPluginConfig.messageCaptureGroupEnd!!,
-            warnPluginConfig.lineCaptureGroup,
-            warnPluginConfig.linePlaceholder!!,
-            warnPluginConfig.messageCaptureGroup!!,
-            index + 1,
-            file,
-            linesFile,
-        )
-        with(warnPluginConfig) {
-            line.extractWarning(
-                generalConfig.expectedWarningsPattern!!,
-                file.name,
-                newLineAndMessage?.first,
-                newLineAndMessage?.second,
-                columnCaptureGroup,
-                benchmarkMode!!,
+        originalPaths: List<Path>,
+        originalPath: Path,
+    ): List<Warning> = if (warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF) {
```

Comment:
may be create `when` here for the future?

Reply:
Changed